File: Behavioural-Patterns/ChainOfResponsibility/Java/Implementations/AuthenticationHandler.java
package Implementations;

import Interfaces.BaseHandler;

public class AuthenticationHandler extends BaseHandler {

    @Override
    protected boolean processRequest(Request request) {
        System.out.println("-> AuthenticationHandler: Checking authentication...");

        if("valid-token".equals(request.getHeader("Authorization"))){
            System.out.println("Authentication successful!");
            return true;
        }
        else{
            System.out.println("Authentication FAILED. Request rejected.");
            return false;
        }
    }
    
}

File: Behavioural-Patterns/ChainOfResponsibility/C#/ChainOfResponsibility/Implementations/AuthenticationHandler.cs
public class AuthenticationHandler : BaseHandler {

    protected override bool processRequest(Request request) {
        Console.WriteLine("-> AuthenticationHandler: Checking authentication...");

        if("valid-token".Equals(request.getHeader("Authorization"))){
            Console.WriteLine("Authentication successful!");
            return true;
        }
        else{
            Console.WriteLine("Authentication FAILED. Request rejected.");
            return false;
        }
    }
    
}
